python/pyarrow/fs.py

```diff
@@ -26,3 +26,20 @@
     LocalFileSystemOptions,
     SubTreeFileSystem
 )
+
+try:
+    from pyarrow._hdfs import HdfsOptions, HadoopFileSystem  # noqa
```

Comment:
`hdfs.py` is already taken by the previous hdfs API, so I cannot expose hdfs in a similar manner then we did with S3. IMO it would be the best to expose these classes under the `pyarrow.fs` namespace, but I also understand that importing optional modules is error prone. I think we should resolve this error-prone-ness by hardcoding the built features to pyarrow somehow. I'm going to add a JIRA about it.

Reply:
@pitrou I've removed the not yet released s3fs.py in favor of `pyarrow.fs`

Reply:
+1.